doc/advanced-topics.rst

```diff
@@ -201,6 +201,124 @@ order of operation if splash screen is enabled:
    resumed.
 
 
+.. _bootloader environment variables:
+
+Environment Variables Used by Frozen Applications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+At run-time, PyInstaller's bootloader uses several environment variables.
+Broadly speaking, these can be grouped into two categories: *public*
+environment variables that allow user or application developer to
+alter the behavior of the frozen application, and *private* environment
+variables, which are used internally by the bootloader for inter-process
+communication (for example, so that onefile parent process can communicate
+the location of the temporary directory with extracted files to the child
+process).
+
+.. Note::
+   PyInstaller-frozen applications use isolated instances of embedded
+   python interpreter, and as such, do **not** process `environment variables
+   used by python interpreter <https://docs.python.org/3/using/cmdline.html#environment-variables>`_.
+   Some of these options can be (permanently) set at build time; see
+   :ref:`specifying python interpreter options`.
+
+Public environment variables
+----------------------------
+
+Public environment variables have ``PYINSTALLER_`` prefix. Currently,
+the following variables are used:
+
+* ``PYINSTALLER_SUPPRESS_SPLASH_SCREEN``: setting this environment variable
+  to 1 allows user to suppress splash screen in a splash-screen enabled
+  application. This causes splash screen not to be shown, and functions
+  of :mod:`pyi_splash` module become no-op without raising errors or warnings.
```

Comment:
There is a proper Sphinx cross-referencable API reference way of documenting environment variables.

```rst
.. envvar:: PYINSTALLER_SUPPRESS_SPLASH_SCREEN

  setting this environment variable
  to 1 allows user to suppress splash screen in a splash-screen enabled
  application. This causes splash screen not to be shown, and functions
  of :mod:`pyi_splash` module become no-op without raising errors or warnings.
```
Not sure it actually looks any better though.

Reply:
I like it. Plus, it allows for cross-referencing from other parts of the docs.